Gem/Code/Source/RHI/XRExampleComponent.cpp

```diff
@@ -0,0 +1,377 @@
+        if (xrSystem && xrSystem->ShouldRender())
+        {
+            AZ::RPI::FovData fovData = xrSystem->GetViewFov(m_viewIndex);
+            AZ::RPI::PoseData poseData = xrSystem->GetViewPose(m_viewIndex);
+                
+            static const float clip_near = 0.05f;
+            static const float clip_far = 100.0f;
+            projection = xrSystem->CreateProjectionOffset(fovData.m_angleLeft, fovData.m_angleRight, 
+                                                          fovData.m_angleDown, fovData.m_angleUp, 
+                                                          clip_near, clip_far);
+
+            AZ::Matrix4x4 viewMat = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(poseData.orientation, poseData.position);;
+            viewMat.InvertTransform();
+            m_viewProjMatrix = projection * viewMat;
+ 
+            AZ::Matrix4x4 initialScaleMat = AZ::Matrix4x4::CreateScale(AZ::Vector3(0.1f, 0.1f, 0.1f));
+
+            //Model matrix for the cube related to the front view
+            AZ::RPI::PoseData frontViewPoseData = xrSystem->GetViewFrontPose();
+            m_modelMatrices[0] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(frontViewPoseData.orientation, frontViewPoseData.position) * initialScaleMat;
+                      
+            //Model matrix for the cube related to the left controller
+            AZ::RPI::PoseData controllerLeftPose = xrSystem->GetControllerPose(0);
+            AZ::Matrix4x4 leftScaleMat = initialScaleMat * AZ::Matrix4x4::CreateScale(AZ::Vector3(xrSystem->GetControllerScale(0), xrSystem->GetControllerScale(0), xrSystem->GetControllerScale(0)));
+            m_modelMatrices[1] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(controllerLeftPose.orientation, controllerLeftPose.position) * leftScaleMat;
+
+            //Model matrix for the cube related to the right controller
+            AZ::Matrix4x4 rightScaleMat = initialScaleMat * AZ::Matrix4x4::CreateScale(AZ::Vector3(xrSystem->GetControllerScale(1), xrSystem->GetControllerScale(1), xrSystem->GetControllerScale(1)));
+            AZ::RPI::PoseData controllerRightPose = xrSystem->GetControllerPose(1);
+            m_modelMatrices[2] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(controllerRightPose.orientation, controllerRightPose.position) * rightScaleMat;
+        }      
+        
+        for (int i = 0; i < s_numberOfCubes; ++i)
+        {
+            m_shaderResourceGroups[i]->SetConstant(m_shaderIndexWorldMat, m_modelMatrices[i]);
+            m_shaderResourceGroups[i]->SetConstant(m_shaderIndexViewProj, m_viewProjMatrix);
+            m_shaderResourceGroups[i]->Compile();
+        }
+
+        BasicRHIComponent::OnFramePrepare(frameGraphBuilder);
+    }
+
+    XRExampleComponent::SingleCubeBufferData XRExampleComponent::CreateSingleCubeBufferData([[maybe_unused]] const AZ::Vector4 color)
+    {
+        AZStd::vector<AZ::Vector4> vertexColor =
```

Comment:
`vertexColor` can be const.

Also, since the size will always be the same this can just be a `fixed_vector`, it will deduce the same from the initialization and not heap memory will be used.